File: 100-same-tree.py
import sys

if len(sys.argv) < 2:
    print("- Parameter(s) expected.")
    sys.exit(1)


# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def isSameTree(self, p, q):
        if not p and not q:
            return True
        if not p or not q:
            return False
        if p.val != q.val:
            return False
        return self.isSameTree(p.left, q.left) and self.isSameTree(p.right, q.right)


if __name__ == "__main__":
    import sys

    if len(sys.argv) != 7:
        print("Usage: python same_tree.py p_val1 p_val2 q_val1 q_val2 q_val3 q_val4")
    else:
        # Construct the trees from command-line arguments
        p_val1, p_val2, q_val1, q_val2, q_val3, q_val4 = map(int, sys.argv[1:])
        p = TreeNode(p_val1)
        p.left = TreeNode(p_val2)
        q = TreeNode(q_val1)
        q.left = TreeNode(q_val2)
        q.right = TreeNode(q_val3)
        q.right.right = TreeNode(q_val4)

        solution = Solution()
        result = solution.isSameTree(p, q)
        print(p)
        print(q)
        if result:
            print("The given trees are identical.")
        else:
            print("The given trees are not identical.")

File: 100-same-tree.js
// LeetCode Problem 100: Same Tree
// Given two binary trees, write a function to check if they are the same or not.
// Two binary trees are considered the same if they are structurally identical and the nodes have the same value.

/**
 * @param {TreeNode} p
 * @param {TreeNode} q
 * @return {boolean}
 */

class TreeNode {
    constructor(val) {
        this.val = val;
        this.left = this.right = null;
    }
}

function isSameTree(p, q) {
    if (!p && !q) {
        return true; // Both trees are empty, so they are the same
    }
    if (!p || !q || p.val !== q.val) {
        return false; // One of the trees is empty or their values are different
    }
    return isSameTree(p.left, q.left) && isSameTree(p.right, q.right);
}

// Read input from command line arguments
const tree1 = parseTreeFromCommandLine(process.argv[2]);
const tree2 = parseTreeFromCommandLine(process.argv[3]);

function parseTreeFromCommandLine(input) {
    // Parse input and construct the tree (you can customize this part)
    // For example, input format: "1,2,null,null,3,null,null"
    const values = input.split(',').map(val => (val === 'null' ? null : parseInt(val)));
    return buildTree(values, 0);
}

function buildTree(values, index) {
    if (index >= values.length || values[index] === null) {
        return null;
    }
    const node = new TreeNode(values[index]);
    node.left = buildTree(values, 2 * index + 1);
    node.right = buildTree(values, 2 * index + 2);
    return node;
}

console.log(isSameTree(tree1, tree2)); // Output: true or false